common/buildcraft/builders/snapshot/Blueprint.java

```diff
@@ -97,9 +98,6 @@ public void deserializeNBT(NBTTagCompound nbt) throws InvalidInputDataException
         NBTTagList list = nbt.hasKey("data", Constants.NBT.TAG_LIST) ? nbt.getTagList("data", Constants.NBT.TAG_INT)
             : null;
         int[] serializedData = nbt.hasKey("data", Constants.NBT.TAG_INT_ARRAY) ? nbt.getIntArray("data") : new int[0];
-        if (serializedData == null && list == null) {
-            throw new InvalidInputDataException("Can't read a blueprint with no data!");
-        }
```

Comment:
it simply can't be null, if you look at the 2 this is data that already checked in the 2 lines above with nbt.hasKey and a ternairy if